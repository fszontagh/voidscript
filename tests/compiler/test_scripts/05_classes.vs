// Test basic class definitions and usage
class Calculator {
    private: int $value = 0;
    
    function setValue(int $val) {
        $this->value = $val;
    }
    
    function getValue() int {
        return $this->value;
    }
    
    function add(int $num) int {
        $this->value = $this->value + $num;
        return $this->value;
    }
    
    function multiply(int $num) int {
        $this->value = $this->value * $num;
        return $this->value;
    }
}

// Create instance and test methods
Calculator $calc = new Calculator();
$calc->setValue(10);

printnl("Initial value: ", $calc->getValue());

int $result1 = $calc->add(5);
printnl("After adding 5: ", $result1);

int $result2 = $calc->multiply(2);
printnl("After multiplying by 2: ", $result2);

printnl("Final value: ", $calc->getValue());