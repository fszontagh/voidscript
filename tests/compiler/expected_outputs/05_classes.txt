Initial value: 10
After adding 5: 15
After multiplying by 2: 30
Final value: 30